### ISSUE TITLE
manifest relative URL issue

### ISSUE THREAD

**Issue by dickhardt** [fedidcg/FedCM#266]
reading the manifest specifications more closely, the endpoints are all specified as relative URLs rather than full URLs (which is how the URLs are specified in the OIDC discovery docs)

This presumes the IdP discovery docs and the endpoints are on the same domain. They are not for Hellō -- https://issuer.hello.coop is the IdP `iss` identifier -- but the token endpoints are on a different host -- https://consent.hello.coop

**Comment by samuelgoto** [comment]
ah yeah, that seems like a bug ... off the top of my head, i don't think there would be any issue making these absolute urls (as long as they are under the same eTLD+1, I think that should be fine) as opposed to relative urls.

@cbiesinger @yi-gu any concerns?

If not, I can send a PR to fix the spec and kick off implementation bugs.

**Comment by cbiesinger** [comment]
They need to be same-origin right now. We could perhaps relax that check to eTLD+1 but that is a concept that the web rarely uses.

**Comment by cbiesinger** [comment]
(However absolute URLs do work if they are same-origin)

**Comment by dickhardt** [comment]
Why the requirement to be same-origin, or even eTLD+1? That requirement does not exist for the OIDC discovery document. 
Following security best practices, I have a separation of concerns -- metadata is on different host than the APIs. (they happen to be eTLD+1 == but that is not an OIDC requirement)

**Comment by dickhardt** [comment]
@cbiesinger -- we are referring to what the spec SHOULD be in this discussion, rather than the current implementation, correct?

**Comment by cbiesinger** [comment]
Indeed, we are. But this is the web platform and fedcm opens up cross-origin data transfer, so we need to be careful in how we approach this.

For a start, we would need to make sure to run CSP checks on each redirect. I am also not sure how to feel about the fact that each endpoint could be on a different domain; that seems worrisome.

**Comment by cbiesinger** [comment]
Sorry, I got mixed up in that comment with redirects vs cross-origin direct URLs. But the issues are similar.

**Comment by dickhardt** [comment]
From a security perspective, cookies etc. would still only be shared per the URLs. IE, if `iss=https://issuer.example` and `id_token_endpoint=https://account.example/fed-cm/token` => only cookies that would be sent to  `https://account.example/fedcm/token` would be sent, not cookies for `https://issuer.example` 

GIven the privacy focus of FedCM -- I agree we need to think through if cross domain support could be used/abused for cross site tracking.

**Comment by hlflanagan** [comment]
I propose closing this issue for now. If the original poster, any of the commenters, or someone in the WG or CG wants to advocate for keeping it open and help move it forward, please comment within two weeks of this label being applied.

**Comment by hlflanagan** [comment]
Closing. We can always re-open if we have someone willing to champion the issue.
